<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ALLSort</class>
 <widget class="QMainWindow" name="ALLSort">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>847</width>
    <height>376</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>847</width>
    <height>376</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>847</width>
    <height>376</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>ALLSort</string>
  </property>
  <property name="windowIcon">
   <iconset theme="accessories-calculator">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTextBrowser" name="sorted_array">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>471</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
   </widget>
   <widget class="QTextEdit" name="array_show">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>471</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
   </widget>
   <widget class="QLabel" name="time_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>55</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>55</width>
      <height>21</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>55</width>
      <height>21</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Cascadia Code</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>Time:   </string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>10</y>
      <width>71</width>
      <height>61</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>831</width>
      <height>31</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_arr_crt">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Create array</string>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>10</y>
      <width>31</width>
      <height>351</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QTextBrowser" name="step_log">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>35</y>
      <width>321</width>
      <height>311</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>321</width>
      <height>311</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>320</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
   </widget>
   <widget class="QLabel" name="sort_time">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>330</y>
      <width>411</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>411</width>
      <height>21</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>411</width>
      <height>21</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Cascadia Code</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>180</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="check_sort">
    <property name="geometry">
     <rect>
      <x>439</x>
      <y>180</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Check</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>461</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="array_from_file">
       <property name="minimumSize">
        <size>
         <width>31</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>41</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>File</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="rand_array_button">
       <property name="minimumSize">
        <size>
         <width>111</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>33</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>Randomize array</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="random_range">
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>Â±</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="array_size_box">
       <property name="minimumSize">
        <size>
         <width>51</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>51</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximum">
        <number>5000000</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="tack_var10">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>V10</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="tack_var11">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>V11</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="tack_var12">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>V12</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="tack_var13">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>V13</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="tack_var14">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>35</width>
         <height>31</height>
        </size>
       </property>
       <property name="text">
        <string>V14</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_step_log">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>-10</y>
      <width>261</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>261</width>
      <height>51</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>261</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>Step log</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>411</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QPushButton" name="selection_sort">
       <property name="minimumSize">
        <size>
         <width>86</width>
         <height>30</height>
        </size>
       </property>
       <property name="text">
        <string>Selection sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="shell_sort">
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Shell sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="quick_sort">
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>70</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Quick sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="merge_sort">
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>70</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Merge sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="counting_sort">
       <property name="minimumSize">
        <size>
         <width>70</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>90</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Counting sort</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>471</width>
      <height>21</height>
     </rect>
    </property>
    <property name="value">
     <number>24</number>
    </property>
   </widget>
   <widget class="QCheckBox" name="log_check">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>0</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Log</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
